### PR TITLE
fix: Skip replication slot reconciliation if fenced

### LINKED ISSUE #2784: Skip replication slot synchronization on a fenced instance
If an instance is fenced, the log fills with the following lines:

```
{"level":"info","ts":"2023-09-13T14:51:23Z","logger":"Replicator","msg":"synchronizing replication slots","logging_pod":"kubecon-na-23-eu-central-1-2","err":"getting replication slot status from local: failed to connect to `host=/controller/run user=postgres database=postgres`: dial error (dial unix /controller/run/.s.PGSQL.5432: connect: no such file or directory)"}
```

We should be smarter, and prevent that from happening - similarly to what we do with metrics collection, which correctly reports:

``` 
{"level":"info","ts":"2023-09-13T14:58:22Z","msg":"metrics collection skipped due to fencing","logging_pod":"kubecon-na-23-eu-central-1-2"}
```

### PR DESCRIPTION
**Pull request by YanniHu1996**
closes #2784

### CODE REVIEW

**Review by armru — APPROVED** [review]

File: internal/management/controller/slots/runner/runner.go [hunk 1]
```diff
@@ -98,6 +98,13 @@ func (sr *Replicator) reconcile(ctx context.Context, config *apiv1.ReplicationSl
 		}
 	}()
 
+	contextLog := log.FromContext(ctx)
+
+	if sr.instance.IsFenced() {
+		contextLog.Debug("Replication slots reconciliation skipped: instance is fenced.")
```
Comment (mnencia) [inline internal/management/controller/slots/runner/runner.go:104]: I would set it to `Trace`

Comment (litaocdl) [reply internal/management/controller/slots/runner/runner.go:104]: yeah, trace is fine, let me change

**Review by litaocdl — APPROVED** [review]

**Pull request merged**